indico/modules/categories/controllers/display.py

```diff
@@ -263,12 +263,16 @@ def _process(self):
         # If there are no events in this range, it will include the last and next month containing events.
         past_threshold = self.now - relativedelta(months=1, day=1, hour=0, minute=0)
         future_threshold = self.now + relativedelta(months=1, day=1, hour=0, minute=0)
+        hidden_event_ids = tuple([event.id for event in Event.get_hidden_events(self.category.id)
```

Comment:
Better use a set instead of a tuple here; and move this method/logic into the Category model as you act on a category anyway.

    hidden_event_ids = {e.id for e in self.category.get_hidden_events()}

Reply:
OK, so you want me to also move the can_display(user) check inside the get_hidden_events(), correct?

Reply:
I think it makes sense, unless doing the check outside allows you to do some optimization... But right now I can't think of anything in particular, except that you could skip the call to this altogether in case the user is a category manager (this avoids the event ACL check for non-admin users since if someone is a category manager they are allowed to see everything anyway)

Reply:
OK, I'm fine with it.